go-controller/pkg/ovn/egressfirewall_dns.go

```diff
@@ -178,6 +189,6 @@ func (e *EgressDNS) Shutdown() {
 	close(e.stopChan)
 }
 
-func (e *EgressDNS) signalAdded(dnsName string) {
-	e.added <- dnsName
+func (e *EgressDNS) signalAdded() {
```

Comment:
You want to be a bit more clever here, since you don't need to signal 5 times in a row if you're processing an update.

So, instead, use this trick: https://github.com/kubernetes/kubernetes/blob/710865511ea173bb8a73158a002ff852d097e53b/pkg/util/async/bounded_frequency_runner.go#L215

Reply:
What it does is only send if the channel is empty. That's exactly what you want here.

Reply:
@squeed I think we need to signal everytime, or we could miss a dnsName with a small ttl while a dnsName with a larger ttl was processing.

Reply:
But this is the `added` chan, right? This only signals to re-evaluate the deadline, AIUI...

How could this cause us to miss a refresh?

Reply:
you are right